### ISSUE TITLE
TypeError: Cannot read property 'includes' of undefined

### ISSUE THREAD

**Issue by anik3tra0** [naugtur/blocked-at#3]
### Node Version - 8.5.0

1. Getting this TypeError when I am running this not sure how and why..?
2. Could you provide some usage examples?
3. Any specific reason as to why the default threshold is set at 20ms?

### Code Implementation

```js
let result = [];

blocked((time, stack) => {
  result = result.concat(stack);
  console.log('engine.js blocked', time, stack);
}, { threshold: 20 });
```

### Below is my log when I start my app.
```sh
engine.js blocked 23.87353503417969 [ '    at process.nextTick (internal/process/next_tick.js:270:7)',
  '    at handleOperationCallback (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:468:24)',
  '    at /home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:504:9',
  '    at authenticateStragglers (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:443:16)',
  '    at Connection.messageHandler (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:477:5)',
  '    at Socket.<anonymous> (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/connection.js:331:22)',
  '    at emitOne (events.js:115:13)',
  '    at Socket.emit (events.js:210:7)',
  '    at addChunk (_stream_readable.js:266:12)',
  '    at readableAddChunk (_stream_readable.js:253:11)',
  '    at Socket.Readable.push (_stream_readable.js:211:10)',
  '    at TCP.onread (net.js:585:20)' ]
engine.js blocked 24.23693408203125 [ '    at process.nextTick (internal/process/next_tick.js:270:7)',
  '    at handleOperationCallback (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:468:24)',
  '    at /home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:504:9',
  '    at authenticateStragglers (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:443:16)',
  '    at Connection.messageHandler (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/pool.js:477:5)',
  '    at Socket.<anonymous> (/home/deploy/apps/syook-tnt-server/node_modules/mongodb-core/lib/connection/connection.js:331:22)',
  '    at emitOne (events.js:115:13)',
  '    at Socket.emit (events.js:210:7)',
  '    at addChunk (_stream_readable.js:266:12)',
  '    at readableAddChunk (_stream_readable.js:253:11)',
  '    at Socket.Readable.push (_stream_readable.js:211:10)',
  '    at TCP.onread (net.js:585:20)' ]
engine.js blocked 24.492054931640624 [ '    at internalConnect (net.js:939:26)',
  '    at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:1098:7)',
  '    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:98:10)' ]
engine.js blocked 24.613239013671876 [ '    at internalConnect (net.js:939:26)',
  '    at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:1098:7)',
  '    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:98:10)' ]
setTCPConnection true
new client connection from ::ffff:43.228.229.58:38268
connection data from ::ffff:43.228.229.58:38268: "|2017-09-23 16:09:40,D4:22:01:00:03:FD,01,001,00,-69,100,|2017-09-23 16:09:40,D4:22:01:00:03:F7,01,003,00,-88,100,|2017-09-23 16:09:40,D4:22:01:00:03:F3,02,001,00,-98,100,003,00,-88,100,|2017-09-23 16:09:40,D4:22:01:00:03:FC,02,001,00,-78,100,003,00,-94,100,|2017-09-23 16:09:40,D4:22:01:00:03:F2,01,003,00,-70,100,|2017-09-23 16:09:40,D4:22:01:00:03:F4,02,001,00,-89,100,003,00,-88,100,|2017-09-23 16:09:40,D4:22:01:00:03:F6,02,001,00,-87,100,003,00,-88,100,|2017-09-23 16:09:40,D4:22:01:00:03:FF,01,001,00,-47,100,|2017-09-23 16:09:40,D4:22:01:00:03:EB,01,001,00,-49,100,|2017-09-23 16:09:40,D4:22:01:00:03:EA,01,001,00,-58,100,|2017-09-23 16:09:40,D4:22:01:00:03:FA,01,001,00,-64,100,|2017-09-23 16:09:40,D4:22:01:00:03:E9,01,001,00,-49,100,|2017-09-23 16:09:40,D4:22:01:00:03:ED,01,001,00,-79,100,|2017-09-23 16:09:40,D4:22:01:00:03:EE,01,001,00,-79,100,|2017-09-23 16:09:40,D4:22:01:00:03:F9,01,001,00,-58,100,|2017-09-23 16:09:40,D4:22:01:00:04:00,01,001,00,-77,100,|2017-09-23 16:09:40,D4:22:01:00:03:F0,01,001,00,-59,100,"
data contains gatewayHandshakeMsg false
mode =  IDLE
info: [Stream Log] D4:22:01:00:03:FD reader pings 001 00 -69 100
info: [Stream Log] D4:22:01:00:03:F7 reader pings 003 00 -88 100
info: [Stream Log] D4:22:01:00:03:F3 reader pings 001 00 -98 100
info: [Stream Log] D4:22:01:00:03:F3 reader pings 003 00 -88 100
info: [Stream Log] D4:22:01:00:03:FC reader pings 001 00 -78 100
info: [Stream Log] D4:22:01:00:03:FC reader pings 003 00 -94 100
info: [Stream Log] D4:22:01:00:03:F2 reader pings 003 00 -70 100
info: [Stream Log] D4:22:01:00:03:F4 reader pings 001 00 -89 100
info: [Stream Log] D4:22:01:00:03:F4 reader pings 003 00 -88 100
info: [Stream Log] D4:22:01:00:03:F6 reader pings 001 00 -87 100
info: [Stream Log] D4:22:01:00:03:F6 reader pings 003 00 -88 100
info: [Stream Log] D4:22:01:00:03:FF reader pings 001 00 -47 100
info: [Stream Log] D4:22:01:00:03:EB reader pings 001 00 -49 100
info: [Stream Log] D4:22:01:00:03:EA reader pings 001 00 -58 100
info: [Stream Log] D4:22:01:00:03:FA reader pings 001 00 -64 100
info: [Stream Log] D4:22:01:00:03:E9 reader pings 001 00 -49 100
info: [Stream Log] D4:22:01:00:03:ED reader pings 001 00 -79 100
info: [Stream Log] D4:22:01:00:03:EE reader pings 001 00 -79 100
info: [Stream Log] D4:22:01:00:03:F9 reader pings 001 00 -58 100
info: [Stream Log] D4:22:01:00:04:00 reader pings 001 00 -77 100
info: [Stream Log] D4:22:01:00:03:F0 reader pings 001 00 -59 100
TypeError: Cannot read property 'includes' of undefined
    at cleanStack (/home/deploy/apps/syook-tnt-server/node_modules/blocked-at/index.js:9:38)
    at AsyncHook.after (/home/deploy/apps/syook-tnt-server/node_modules/blocked-at/index.js:42:28)
    at emitAfterNative (async_hooks.js:375:40)
```

**Comment by naugtur** [comment]
1. looking into it
2. Your usage is correct except for the result variable which was only useful in tests. 
All you need to do is pass a function to `blocked-at` and log the time and stack
3. In my testing parts of core or regular functions were exceeding the 10ms threshold which used to be the default in good old `blocked` package. It's an arbitrary default. Feel free to use a higher number, it's not important for it to be 20ms

**Comment by naugtur** [comment]
About 1. 
I see it must have been a really unusual stacktrace. 

I've pushed an updated version to a branch, please try 
```
npm install naugtur/blocked-at#stack-edgecase
```

It should work fine.

Also, if you pass a `debug:true` option to that version, it will print the stack and a lot of other information I can use to refine the implementation. If you have time to try it, I'd be grateful.

**Comment by anik3tra0** [comment]
Point #1 is now resolved and working after I installed the package/branch.
Point #3 - Here is what I could log by setting  `debug: true` as an option. Hope it helps. If you need more of these I can log them and send across.

```sh
warn: [Process Blockage Log] index.js was blocked 132.845452999115
init 3210
init 3210
after 3205
after 3205
before 3207
before 3207
after 3207
after 3207
before 3208
before 3208
after 3208
after 3208
before 3209
before 3209
after 3209
after 3209
before 3210
before 3210
after 3210
after 3210
before 44
before 44
after 44
after 44
before 3030
before 3030
init 3211
init 3211
after 3030
after 3030
before 3206
before 3206
init 3212
init 3212
after 3206
after 3206
before 3211
before 3211
after 3211
after 3211
before 3212
before 3212
after 3212
after 3212
before 2816
before 2816
after 2816
after 2816
before 46
before 46
before 3025
before 3025
after 3025
after 3025
after 46
after 46
before 3026
before 3026
after 3026
after 3026
before 46
before 46
before 45
before 45
init 3213
init 3213
init 3214
init 3214
init 3215
init 3215
after 45
after 45
after 46
after 46
before 44
before 44
after 44
after 44
```
This works so you can close this issue and release a fix. Your call.

**Comment by naugtur** [comment]
I'll release in a few hours. 

The log you posted doesn't have the stack I hoped for unfortunately. 

If you'd be so kind and run it a few times until you get a debug with something that resembles a stacktrace, would be great. If you can't reproduce the stacktrace, I'll have to wait for the next case.

**Comment by naugtur** [comment]
Fixed in v1.0.3
